juju/client/proxy/factory.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from juju.client.proxy.kubernetes.proxy import KubernetesProxy
+
+class ProxyTypeNotDefinedError(Exception):
+    pass
+
+class UnknownProxyTypeError(Exception):
```

Comment:
It may make sense just to use the built-in `ValueError` exception type for this. That's typically what's raised for invalid params to a function, and callers probably don't need the extra specificity. If you really want/need to keep it, make it inherit from `ValueError`.